src/router/routes.js

```diff
@@ -442,15 +442,24 @@ export default [
 			if (!hash || to?.query?.goTo) return next();
 
 			const hashValue = hash.slice(1);
+			const [goTo, hashQueryString] = hashValue.split('?');
+
+			const newQuery = { ...to.query, goTo };
```

Comment:
Are UTM params also maintained like expected?

Reply:
Yes, I've been testing with `/mykiva#mykiva-borrower-carousel?a=1&b=2&c=3&utm_campaing=test` and this change makes the final url be `?goTo=mykiva-borrower-carousel&a=1&b=2&c=3&utm_campaign=test`